src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DataAnnotationsModelMetadataProvider.cs

```diff
@@ -17,11 +23,21 @@ public class DataAnnotationsModelMetadataProvider : AssociatedMetadataProvider<C
             return new CachedDataAnnotationsModelMetadata(this, containerType, modelType, propertyName, attributes);
         }
 
+        /// <inheritdoc />
+        /// <remarks>
+        /// Copies only a few values from the <paramref name="prototype"/>. Unlikely the rest have been computed.
+        /// </remarks>
         protected override CachedDataAnnotationsModelMetadata CreateMetadataFromPrototype(
             CachedDataAnnotationsModelMetadata prototype,
             Func<object> modelAccessor)
         {
-            return new CachedDataAnnotationsModelMetadata(prototype, modelAccessor);
+            var metadata = new CachedDataAnnotationsModelMetadata(prototype, modelAccessor);
+            foreach (var keyValuePair in prototype.AdditionalValues)
+            {
+                metadata.AdditionalValues.Add(keyValuePair);
+            }
```

Comment:
we don't have a copy constructor for `ModelMetadata`.  I originally placed the `AdditionalValues` copy in the `CachedModelMetadata` copy constructor.  but it looked strange having one copy there and another in the `EmptyModelMetadataProvider.CreateMetadataFromPrototype()` implementation.  so I decided putting the copy in both providers made more sense.